api_backend_server/app.py

```diff
@@ -1,25 +1,160 @@
+    user_id = db.Column(db.Integer)
+    name = db.Column(db.String(50), unique=False, nullable=False)
+
+
+def token_required(f):
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+
+        token = None
+
+        if "x-access-tokens" in request.headers:
+            token = request.headers["x-access-tokens"]
+
+        if not token:
+            return jsonify({"message": "a valid token is missing"})
+
+        try:
+            data = jwt.decode(token, app.config["SECRET_KEY"])
+        except jwt.exceptions.DecodeError:
+            return jsonify({"message": "token is invalid"})
+        else:
+            current_user = Users.query.filter_by(public_id=data['public_id']).first()
+
+        return f(current_user, *args, **kwargs)
+
+    return wrapper
+
 
 @app.route("/", methods=["GET"])
 def index():
     return jsonify(message="Hello from Python backend server")
 
 
-@app.route("/login", methods=["POST"])
-def login():
-    username = request.args.get("username")
-    user_id = randint(0, 10)
-    return jsonify(username=username,
-                   userid=user_id)
+@app.route("/api/register", methods=["POST"])
+def signup_user():
+    data = request.get_json()
+
+    hashed_password = generate_password_hash(data["password"], method="sha256")
+    public_id = str(uuid.uuid4())
+
+    new_user = Users(public_id=public_id, name=data["name"], password=hashed_password, admin=False)
+    db.session.add(new_user)
+    db.session.commit()
+
+    return jsonify({"message": "registered successfuly",
+                    "user_id": public_id})
+
+
+@app.route("/api/login", methods=["POST"])
+def login_user():
+
+    auth = request.authorization
+
+    if not auth or not auth.username or not auth.password:
+        return make_response("could not verify", 401, {"WWW.Authentication": "Basic Reals: 'login required'"})
+
+    user = Users.query.filter_by(name=auth.username).first()
+
+    if check_password_hash(user.password, auth.password):
+        token = jwt.encode(
+                            {
+                              "public_id": user.public_id,
+                              "exp": datetime.datetime.utcnow() + datetime.timedelta(minutes=30)
+                            },
+                            app.config["SECRET_KEY"]
+                          )
+        return jsonify({"token": token.decode("UTF-8")})
+
+    return make_response("could not verify", 401, {"WWW.Authentication": "Basic realm: 'login required'"})
+
+
+@app.route("/api/users", methods=["GET"])
+@token_required
+def get_all_users(current_user):
+
+    users = Users.query.all()
+
+    result = []
+
+    for user in users:
+        user_data = {}
+        user_data["public_id"] = user.public_id
+        user_data["name"] = user.name
+        user_data["password"] = user.password
+        user_data["admin"] = user.admin
+
+        result.append(user_data)
+
+    return jsonify({"users": result})
+
+
+@app.route("/api/create_activity", methods=["POST"])
```

Comment:
Metoda POST mówi już, że to jest endpoint od tworzenia aktywności
```suggestion
@app.route("/api/activities", methods=["POST"])
```